Chartboost/src/main/java/com/mopub/mobileads/ChartboostShared.java

```diff
@@ -103,13 +107,6 @@ public static synchronized boolean initializeSdk(@NonNull Context context,
         final String appId = serverExtras.get(APP_ID_KEY);
         final String appSignature = serverExtras.get(APP_SIGNATURE_KEY);
 
-        if (!TextUtils.isEmpty(appId) && !TextUtils.isEmpty(appSignature)) {
-            if (appId.equals(mAppId) && appSignature.equals(mAppSignature)) {
-                // We don't need to reinitialize.
-                return false;
-            }
-        }
-
```

Comment:
Should we recover the check we removed from the ChartboostAdapterConfiguration here?:
```
if (TextUtils.isEmpty(appId) || TextUtils.isEmpty(appSignature)) {
                         MoPubLog.log(CUSTOM, ADAPTER_NAME, "Chartboost's initialization " +
                                 "succeeded, but unable to call Chartboost's startWithAppId(). " +
                                 "Ensure Chartboost's " + APP_ID_KEY + " and " + APP_SIGNATURE_KEY +
                                 "are populated on the MoPub dashboard. Note that initialization on " +
                                 "the first app launch is a no-op.");
                     }
```

Reply:
The previous behavior was to call Chartboost.startWithAppId even if the appID or appSignature were empty... should we keep that behavior? 
Is there any difference between caching an ad when initialization fails due to appID and appSignature being invalid and caching an ad when initialization was never even tried?